### ISSUE TITLE
[Bug]: Cannot use PG 17 replication slot failover (dbname must be specified in primary_conninfo)

### ISSUE THREAD

**Issue by jfrconley** [cloudnative-pg/cloudnative-pg#5917]
### Is there an existing issue already for this bug?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [X] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [X] I have read the troubleshooting guide and I think this is a new bug.

### Version

1.24.0

### What version of Kubernetes are you using?

1.31

### What is your Kubernetes environment?

Self-managed: k3s

### How did you install the operator?

Helm

### What happened?

I'm trying to test the new failover slots into PG 17. I am able to turn on hot_standby_feedback and sync_replication_slots but the slots never synchronize. The replica is logging an error from the "slotsync worker" that says that dbname must be specified in the primary_conninfo in order to enable failover sync. [According to the PG docs](https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-PRIMARY-CONNINFO) the dbname parameter is used exclusively for slot synchronization, so the issue might be relatively straightforward to solve.

### Relevant log output

```shell
{"level":"info","ts":"2024-10-21T23:57:29.986731099Z","logger":"postgres","msg":"record","logging_pod":"best-cluster-1","record":{"log_time":"2024-10-21 23:57:29.986 UTC","process_id":"4376","session_id":"6716ea69.1118","session_line_num":"2","session_start_time":"2024-10-21 23:57:29 UTC","virtual_transaction_id":"92/0","transaction_id":"0","error_severity":"ERROR","sql_state_code":"22023","message":"replication slot synchronization requires \"dbname\" to be specified in \"primary_conninfo\"","backend_type":"slotsync worker","query_id":"0"}}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 99hops** [comment]
Absolutely! @jfrconley ![Image](https://github.com/user-attachments/assets/849585a8-748e-4f31-a414-7669dbbb8a96)

**Comment by quajones** [comment]
We're also experiencing this issue when attempting to enable logical replication failover for PG 17 instance. Any updates or plans to support this feature?

I took a stab at the required changes within this pr: https://github.com/cloudnative-pg/cloudnative-pg/pull/7298

**Comment by kndoni** [comment]
I have the same issue while trying to enable this
